src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -1,41 +1,95 @@
 import edgir.elem.elem.{BlockLike, HierarchyBlock}
 import org.eclipse.elk.graph.ElkNode
 import edg.wir.DesignPath
 import edg_ide.edgir_graph.HierarchyGraphElk.PropertyMapper
 import edg_ide.edgir_graph.{EdgeWrapper, HierarchyGraphElk, NodeDataWrapper, PortWrapper}
 import edg_ide.swing.blocks.ElkNodePainter
+import edgir.ref.ref.LibraryPath
 
 import java.awt.Color
 import java.io.{FileNotFoundException, FileOutputStream, IOException}
-
+import scala.collection.mutable
 
 object PDFGeneratorUtil{
 
+  /*
+    Note for future development:
+      If infinite loop (and eventually StackOverflow) occurs when adding header/footer, check that the
+      page size is large enough/page margin is small enough so that header/footer content do not exceed
+      the available space in the document.
+   */
+  final val PAGE_MARGIN = 15
+  final val font = new Font(Font.HELVETICA, 12f, Font.UNDERLINE, Color.BLUE)
+  final val TABLE_ROW_HEIGHT = 23f
+
+
   private def generatePageSize(node: ElkNode): (Float, Float) = {
-    val width = node.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val height = node.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
+    val width = node.getWidth.toFloat + 2.5f * ElkNodePainter.margin.toFloat
+    val height = node.getHeight.toFloat + 2.5f * ElkNodePainter.margin.toFloat
     (width, height)
   }
 
+
+  def getDuplicationList(block: HierarchyBlock, path: DesignPath = DesignPath(),
+                         dupList: mutable.Map[LibraryPath, Set[DesignPath]]): Unit ={
+
+    val className = block.getSelfClass
+    dupList.getOrElseUpdate(className, Set(path)) match {
+      case existingSet: Set[DesignPath] => dupList(className) = existingSet + path
+    }
+
+    block.blocks.asPairs.map {
+      case (name, subblock) => (name, subblock.`type`)
+    }.collect {
+      case (name, BlockLike.Type.Hierarchy(subblock)) =>  (path + name, subblock)
+    }.foreach {
+      case (path, subblock) => getDuplicationList(subblock, path, dupList)
+    }
+  }
+
+
   def generate(content: HierarchyBlock,
                mappers: Seq[PropertyMapper[NodeDataWrapper, PortWrapper, EdgeWrapper]] = Seq(),
                fileName: String): Unit = {
+
+    val dupList = mutable.Map[LibraryPath, Set[DesignPath]]()
+    getDuplicationList(content, dupList = dupList)
+
     try {
       val document = new Document()
       val writer = PdfWriter.getInstance(document, new FileOutputStream(fileName))
-      val footer = new HeaderFooter(true)
-      footer.setBorder(Rectangle.NO_BORDER)
-      footer.setAlignment(Element.ALIGN_RIGHT)
-      document.setFooter(footer)
 
-      def printNode(node: ElkNode): Unit = {
+
+      def printNode(node: ElkNode, className: LibraryPath, path: DesignPath): Unit = {
+        val dupSet = dupList.getOrElse(className, Set.empty)
         val (width, height) = generatePageSize(node)
-        document.setPageSize(new Rectangle(width, height))
+        val adjustedHeight = if(dupSet.size == 1) {
```

Comment:
Not a "now" thing, but eventually if we display more data in the table, it might make sense to print the table for all cases, whether the page corresponds to just one, or multiple blocks. It also simplifies the code by eliminating a special case.